include/cvc5/cvc5_proof_rule.h

```diff
@@ -2499,13 +2499,17 @@ enum ENUM(ProofRewriteRule)
   /**
    * \verbatim embed:rst:leading-asterisk
    * **Equality -- Macro lambda application eliminate shadow**
-   * 
+   *
    * .. math::
    *   ((\lambda x_1 \ldots x_n.\> t) \ t_1 \ldots t_n) = ((\lambda y_1 \ldots y_n.\> t \sigma) \ t_1 \ldots t_n)
-   * 
+   *
    * where :math:`\sigma` replaces :math:`x_1, \dots, x_n` by
    * :math:`y_1, \dots, y_n`. The terms may either be of kind
    * `cvc5::Kind::APPLY_UF` or `cvc5::Kind::HO_APPLY`.
+   * This rule ensures that :math:`y_1, \dots, y_n` are not in the free
+   * variables of :math:`t_1 \ldots t_n` and
```

Comment:
it's the free variables of the lambda expression that you need to guard against.

Reply:
The alpha equivalence condition ensures this.

Reply:
This should ensure alpha-equivalence, see my other comment :)